### DISCUSSION TITLE
Auto upload to NAS Possible?

### DISCUSSION THREAD

**Discussion by DaleJP** [gopro/OpenGoPro#171] [Q&A]
Is there any way I can have the cameras at the end of a shoot auto-upload their footage to a folder with this? I don't care if it is scheduled or what. I just want non-techy people to be able to plug it in and it backs up all the files.

I'm already using the API to download the footage after it has been uploaded to GoPro but I would love to have it directly go to my own server to cut out the middleware.

**Comment by druckgott** [comment]
I also seach a way to Auto upload files to my nas from hero 10. Best case without cloud because in my are is no good internet acceess

**Comment by Fougl** [comment]
> Is there any way I can have the cameras at the end of a shoot auto-upload their footage to a folder with this? I don't care if it is scheduled or what. I just want non-techy people to be able to plug it in and it backs up all the files.
> 
> I'm already using the API to download the footage after it has been uploaded to GoPro but I would love to have it directly go to my own server to cut out the middleware.

Hi, Which API are you using to download the footage? Thanks

**Comment by jonericcook** [comment]
@Fougl https://gopro.github.io/OpenGoPro/http_2_0#downloading-media

**Comment by dhiltonp** [comment]
There's now a python api with example code to connect to a camera and download all media: https://github.com/gopro/OpenGoPro/tree/main/demos/python/sdk_wireless_camera_control#usage

https://github.com/gopro/labs/discussions/908#discussioncomment-9443156

There's also a COHN option, where the gopro will automatically connect to wifi, you can use a script to automatically pull the footage; it's not as easy as if it were directly configured on the camera itself, but it is doable.
